api/src/State/ItemStateProcessor.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\State;
+
+use ApiPlatform\Metadata\DeleteOperationInterface;
+use ApiPlatform\Metadata\Operation;
+use ApiPlatform\Metadata\Post;
+use ApiPlatform\State\ProcessorInterface;
+use App\Entity\Item;
+use App\ApiResource\Item as ItemResource;
+use App\Mapper\EntityMapper;
+use App\Repository\ItemRepository;
+
+readonly class ItemStateProcessor implements ProcessorInterface
+{
+    public function __construct(
+        private ItemRepository $itemRepository,
+        private ItemStateProvider $itemProvider,
+        private EntityMapper $entityMapper
+    ) {
+    }
+
+    /**
+     * @throws \ReflectionException
+     */
+    public function process(
+        mixed $data,
+        Operation $operation,
+        array $uriVariables = [],
+        array $context = []
+    ) {
```

Comment:
I'm going to be _that_ ass and say you could probably have a return type here of `object|array|null` since that's what the `ItemProvider` will return, and this either return's the output of that or `null`? I hate myself a little now